src/screens/map-screen.tsx

```diff
@@ -0,0 +1,136 @@
+import { async } from '@firebase/util';
+import { Product } from 'src/cards/product';
+import { BasicProductDataID } from 'src/types/products';
+import { converters } from 'src/firebase/db-converters';
+
+import { fetchCloseData } from 'src/firebase/map-api';
+import { useFetchLocationProductsQuery } from 'src/services/products-queries';
+import { LocationArray } from 'src/types/products';
+import { DistanceProducts } from 'src/types/products';
+import { ActivityIndicator } from 'react-native';
+
+// type Location = [number, number];
+
+export const MapScreen = () => {
+    const [location, setLocation] = useState<LocationArray>();
+    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+    const [product, setProducts] = useState<BasicProductDataID[]>();
+    const [distance, setDistance] = useState<number>(50);
+    const [inputToRTK, setInputToRTK] = useState<DistanceProducts | undefined>();
+    const [distanceLabel, setDistanceLabel] = useState(50);
+
+    const { data, isFetching, isLoading, isError, error, isSuccess, refetch } =
+        useFetchLocationProductsQuery(inputToRTK);
+
+    useEffect(() => {
+        (async () => {
+            let { status } = await Location.requestForegroundPermissionsAsync();
+            if (status !== 'granted') {
+                setErrorMsg('Permission to access location was denied');
+                return;
+            }
+
+            let locationz = await Location.getCurrentPositionAsync({});
+
+            // console.log(location.coords.latitude);
+            const loxz: LocationArray = [locationz.coords.latitude, locationz.coords.longitude];
+
+            setLocation(loxz);
+
+            // console.log(location);
+
+            const datas = {
+                loc: loxz,
+                dis: distance,
+            };
+            setInputToRTK(datas);
+        })();
+
+        // const lat = 51.5074;
+        // const lng = 0.1278;
+        // const hash = geohashForLocation([lat, lng]);
+
+        // console.log(hash);
+
+        // text = JSON.stringify(location);
+
+        // console.log(text.latitude);
+
+        // const datas = {
+        //     loc: location,
+        //     dis: distance,
+        // };
+        // setInputToRTK(datas);
+    }, []);
+
+    useEffect(() => {
+        // getPost();
+        setProducts(data);
+        // console.log(product);
+    }, [data]);
+
+    useEffect(() => {
+        const datas = {
+            loc: location,
+            dis: distance,
+        };
+        setInputToRTK(datas);
+    }, [distance]);
+
+    // const getPost = async () => {
+    //     const thing: any = await fetchCloseData(location);
+    //     console.log(`thing ${thing}`);
+    //     console.log(thing);
+    //     setProducts(thing);
+    //     // console.log(product);
+    // };
+
+    let text = 'Waiting..';
+    if (errorMsg) {
+        text = errorMsg;
+    } else if (location) {
+        text = JSON.stringify(location);
+    }
+
```

Comment:
can do an inline if here, stay away from using a "let" variable as much as often.
Do this:
`const text = errorMsg ? errorMsg : JSON.stringify(location);`